posthog/api/test/test_insight.py

```diff
@@ -1752,6 +1752,18 @@ def test_cancel_running_query(self) -> None:
         response = self.client.post(f"/api/projects/{self.team.id}/insights/cancel", {"client_query_id": f"testid"})
         self.assertEqual(response.status_code, 201, response.content)
 
+    @patch("posthog.decorators.get_safe_cache")
+    def test_including_query_id_does_not_affect_cache_key(self, patched_get_safe_cache) -> None:
+        self._get_insight_with_client_query_id("b3ef3987-b8e7-4339-b9b8-fa2b65606692")
+        self._get_insight_with_client_query_id("00000000-b8e7-4339-b9b8-fa2b65606692")
+
+        assert patched_get_safe_cache.call_count == 2
+        assert patched_get_safe_cache.call_args_list[0] == patched_get_safe_cache.call_args_list[1]
```

Comment:
```suggestion
        first_call_args, second_call_args = patched_get_safe_cache.call_args_list
        self.assertEqual(first_call_args, second_call_args)
```

Q: Is `assert` the way to go in py tests now? IIRC their output was worse than the `self.` methods.

Reply:
I added https://github.com/hjwp/pytest-icdiff because I find all of the python test diffs hard to read 😊 

It depends on you using `assert` over `self.assertBlah`

pytest docs promote `assert` style...

![Screenshot 2022-11-18 at 11 20 58](https://user-images.githubusercontent.com/984817/202694059-cea3543d-2fc6-4be3-9832-b2930c7077e9.png)

Reply:
(arguably for this test it makes little difference because guid A diff'd against guid B is low value but I write assert style out of habit now)

Reply:
The reason it's not obvious that while we do use `pytest` as a test _runner_ most of our tests are actually written using unittest library. It's a mess, but one where having a consistent code style is useful.

All that said - let's carry on here.